Add back simple `redirect_uri_mismatch` instructions

allenporter opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
Add back simple `redirect_uri_mismatch` instructions since many users may be stuck with existing OAuth for Web credentials. This explains the path forward to move to a Desktop App credential. This also happens if you delete the integration and re-add it and don't choose the right authentication type.

This authentication rewrite lands in `2012.12` so this is targeted at `next`.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/59033
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

allenporter committed:
Add back simple `redirect_uri_mismatch` instructions

Since many users may be stuck with existing OAuth for Web credentials, call out the path forward to move to a Desktop
App credential. This also happens if you delete the integration and re-add it and don't choose the right authentication
type.

allenporter commented:
> It seems that this PR is targeted against an incorrect branch. Documentation updates which apply to our current stable release should target the `current` branch. Please change the target branch of this PR to `current` and rebase if needed. If this is documentation for a new feature, please add a link to that PR in your description. (message by DocsTargetBranch)

I fixed the boxes checked. This is for `next`.

klaasnicolaas commented:
Probot gives this message because you did not include a link to a pull request in core 😉

allenporter commented:
> Probot gives this message because you did not include a link to a pull request in core 😉

Thanks. Its follow up additional documentation for a feature i already added docs for, so i've added the original core PR.
